docs/path-resolution.rst

```diff
@@ -0,0 +1,690 @@
+| ``lib/util.sol``        + ``/project/lib/util.sol``                                       |
++-------------------------+-----------------------------------------------------------------+
+| ``token.sol``           + ``/project/token.sol``                                          |
++-------------------------+-----------------------------------------------------------------+
+| ``/tmp/contract.sol``   + ``/project/tmp/contract.sol``                                   |
++-------------------------+-----------------------------------------------------------------+
+
+.. note::
+
+    Base path does not affect paths you specify directly on the command line.
+    It is prepended only to the source unit names that come from imports and from paths used in
+    ``source.urls`` in Standard JSON.
+
+.. note::
+
+    Base path is prepended no matter whether an import contains a relative or an absolute path.
+    This may not be apparent because the default value of the option is an empty path.
+
+.. note::
+
+    If you set base path to a relative path, it is interpreted as relative to the current working directory.
+    Note that if you do this, all absolute paths will effectively be converted into relative ones
+    if they go through the default file loader.
+    For example ``import "/project/contract.sol"`` with base path set to ``lib/token`` will result
+    in the file loader looking for ``lib/token/project/contract.sol`` in the current working
+    directory.
+
+.. index:: ! remapping; import, ! import; remapping, ! remapping; context, ! remapping; prefix, ! remapping; target
+.. _import-remapping:
+
+Import Remapping
+================
+
+Base path and relative imports on their own allow you to freely move your project around the
+filesystem but force you to keep all files within a single directory and its subdirectories.
+When using external libraries it is often desirable to keep their files in a separate location.
+To help with that, the compiler provides another mechanism: import remapping.
+
+Remapping allows you to have the compiler replace import path prefixes with something else.
+For example you can set up a remapping so that everything imported from the virtual directory
+``github.com/ethereum/dapp-bin/library`` would actually receive source unit names starting with
+``dapp-bin/library``.
+By setting base path to ``/project`` you could then have the compiler find them in
+``/project/dapp-bin/library``
+
+The remappings can depend on a context, which allows you to configure packages to import,
+e.g. different versions of a library of the same name.
+
+.. warning::
+
+    Information about used remappings is stored in contract metadata so, while they let you avoid
+    changing the source, they cannot be used to achieve reproducible builds.
+    The metadata hash embedded in the bytecode will not be the same if you perform import remapping.
+
+Path remappings have the form of ``context:prefix=target``.
```

Comment:
Is a path remapping something else than an import remapping?

Reply:
It's the same. I'm calling it "import remapping" because it really only applies to paths used in imports. Initially I was confused by "path remapping" and thought that it applies to all paths. I.e. those used on the CLI, in JSON, etc. I think the new name clarifies this a bit.

Reply:
I now see that you just pointed out a single place where I still used the old term. Fixed.